lib/src/theme/color_extensions.dart

```diff
@@ -0,0 +1,283 @@
+/// been modified to use Flutter's HSLColor class instead of the custom one in
+/// the TinyColor lib. The functions from TinyColor re-implemented as Color
+/// extensions here are [brighten], [lighten] and [darken]. They are used
+/// for color calculations in ZetaColorScheme, but also exposed for reuse.
+///
+/// Another frequently used extension is [blend] and [blendAlpha] used to blend
+/// two colors using alpha as a percentage or as an 8-bit int alpha value.
+/// This extension is used to calculate branded surface
+/// colors used by ZetaColorScheme's branded surfaces and for automatic dark
+/// color schemes from a light scheme.
+///
+/// The [getShadeColor] extension is less frequently used and when used,
+/// typically used to color makes colors shades for gradient AppBars, with
+/// default setting to not change black and white.
+extension ZetaColorExtensions on Color {
+  /// {@macro zeta.color.color_to_swatch}
+  ZetaColorSwatch get zetaColorSwatch => ZetaColorSwatch.fromColor(this);
+
+  /// Brightens the color with the given integer percentage amount.
+  /// Defaults to 10%.
+  Color brighten([int amount = 10]) {
+    if (amount <= 0) return this;
+    if (amount > 100) return Colors.white;
+    final Color color = Color.fromARGB(
+      alpha,
+      math.max(0, math.min(255, red - (255 * -(amount / 100)).round())),
+      math.max(0, math.min(255, green - (255 * -(amount / 100)).round())),
+      math.max(0, math.min(255, blue - (255 * -(amount / 100)).round())),
+    );
+    return color;
+  }
+
+  /// Lightens the color with the given integer percentage amount.
+  /// Defaults to 10%.
+  Color lighten([int amount = 10]) {
+    if (amount <= 0) return this;
+    if (amount > 100) return Colors.white;
+    // HSLColor returns saturation 1 for black, we want 0 instead to be able
+    // lighten black color up along the grey scale from black.
+    final HSLColor hsl =
+        this == const Color(0xFF000000) ? HSLColor.fromColor(this).withSaturation(0) : HSLColor.fromColor(this);
+    return hsl.withLightness(math.min(1, math.max(0, hsl.lightness + amount / 100))).toColor();
+  }
+
+  /// Darkens the color with the given integer percentage amount.
+  /// Defaults to 10%.
+  Color darken([int amount = 10]) {
+    if (amount <= 0) return this;
+    if (amount > 100) return Colors.black;
+    final HSLColor hsl = HSLColor.fromColor(this);
+    return hsl.withLightness(math.min(1, math.max(0, hsl.lightness - amount / 100))).toColor();
+  }
+
+  /// Get the right black or white contrasting onColor for a color.
+  ///
+  /// The onColor is white if the color has brightness dark and black if
+  /// the color has brightness light.
+  ///
+  /// Use the function to get black/white color typically used for text and
+  /// icons when they are placed on a surface/background using `color` as its
+  /// background color.
+  Color get onColor => ThemeData.estimateBrightnessForColor(this) == Brightness.light ? Colors.black : Colors.white;
```

Comment:
onColor should come from ZetaColors. Something like cool.90 and white.  The correct values are on the figma

Reply:
I've revised the code to utilize the cool 90 shade and white from the ZetaBaseColors palette.